### PR TITLE
impl for sm3 interaction

### LINKED ISSUE #16: adding sm3 test cases between go lang libs.

### PR DESCRIPTION
**Pull request by SamYuan1990**
Signed-off-by: SamYuan1990 <yy19902439@126.com>

### CODE REVIEW

**Comment by SamYuan1990** [comment]
resolve #18 , add interaction check for sm3 digest between go lang libs

File: azure-pipelines.yml [hunk 1]
```diff
@@ -11,7 +11,32 @@ pool:
...
+    - script: cd $(Pipeline.Workspace); git clone https://github.com/guanzhi/GmSSL.git
+      displayName: Clone GMSSL
+    - script: cd $(Pipeline.Workspace)/GmSSL; SYSTEM=`uname -s` ./config --prefix=/usr/local/gmssl; make; sudo make install_sw  
+      displayName: Config, make and install GMSSL
+    - checkout: self
+    - task: GoTool@0
+      inputs:
+        version: $(goVersion)
+    - script: cd interop; go mod vendor
+    - script: cd pku-gm; go mod vendor
```
Comment (davidkhala) [inline azure-pipelines.yml:34]: why we need to cd pku-gm?

File: interop/sm3Interop_test.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+const base_format = "2006-01-02 15:04:05"
+
+func TestSM3(t *testing.T) {
+	// generate a random string as data
+	time := time.Now()
+	str_time := time.Format(base_format)
+	msg := []byte(str_time)
+	// generate key from tj
+	tj_digest := tj.Sm3Sum([]byte(str_time))
+	fmt.Println(tj_digest)
```
Comment (davidkhala) [inline interop/sm3Interop_test.go:22]: I think it is debug purpose code. We could move it into error print section or remove it.

File: interop/sm3Interop_test.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	ccs_digest := ccs.SumSM3(msg)
+	fmt.Println(ccs_digest)
```
Comment (davidkhala) [inline interop/sm3Interop_test.go:24]: same as above

File: interop/sm3Interop_test.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+	sm3hash := pku.New()
+	sm3hash.Write(msg)
+	pku_digest:=sm3hash.Sum(nil)
+	fmt.Println(pku_digest)
```
Comment (davidkhala) [inline interop/sm3Interop_test.go:28]: same as above

File: azure-pipelines.yml (hunk 1 above)
Comment (SamYuan1990) [reply azure-pipelines.yml]: 我发现其他的比如验证key导入的job也有这个问题，我一并修了。

**Comment by SamYuan1990** [comment]
@davidkhala , 我刚刚更新了，麻烦看下。

**Review by davidkhala — APPROVED** [review]
LGTM, 不过对于goland版本我改变了一些看法，所以之后会有一些改动

**Pull request merged**